### ISSUE TITLE
invalid operation

### ISSUE THREAD

**Issue by stuaxo** [01AutoMonkey/open.gl-tutorials-to-pyglet#1]
Hi,
   I got this weird error trying to run example 2-1:

```
$ python 2-1-Drawing_Polygons-Triangle.py 
Traceback (most recent call last):
  File "2-1-Drawing_Polygons-Triangle.py", line 78, in <module>
    glUseProgram(shaderProgram)
  File "/mnt/data/home/stu/.virtualenvs/3d/local/lib/python2.7/site-packages/pyglet-1.2.3-py2.7.egg/pyglet/gl/lib.py", line 104, in errcheck
    raise GLException(msg)
pyglet.gl.lib.GLException: invalid operation

```

**Comment by map01ASCIIDOC10** [comment]
If you're running unaltered code then it seems to be due to the code not being compatible with your setup, or some funk regarding pyglet, because it runs fine on my computer which is:
- Python 2.7.9
- Pyglet 1.2.3
- GPU: NVIDIA Corporation GK107M - GeForce GT 750M (rev a1) / GeForce GT 750M/PCIe/SSE2
- Driver: NVIDIA binary driver - version 346.59
- OS: Ubuntu 15.04 64-bit
- Kernel: Linux 3.19.0-25-generic

You could try out different pyglet versions, make sure you got the latest GPU drivers, make sure your GPU supports at least OpenGL 3.2, and if nothing works you could also try asking on [the Pyglet mailing list](https://groups.google.com/forum/#!forum/pyglet-users).

**Comment by stuaxo** [comment]
Hi,
    It seems to be because I'm running the mesa drivers.   I just wrote some test code pyglet seems to come back with a 3.0 context, even if I ask it for 3.2.

Changing the shaders to #version 130 made everything work :)

Everything else in my setup seems to be the same, apart from Nouveau with 650M and an Intel card (pyglet, python versions are fine etc).

I'll have a further look, since I've been able to get contexts using GLFW, not pyglet.  

TLDR - Change the shader version to 130.

**Comment by stuaxo** [comment]
I had a bit more of a poke at this, and you can get pyglet to give a newer opengl context and work with 150 core shaders:

```
window = pyglet.window.Window(800, 600, "OpenGL", config=Config(major_version=3, minor_version=2))
print window.context.get_info().get_version()
```

_however_ it seems like pyglet isn't fully compatibly with opengl core context as it does some stuff with older gl in its resize method:

```
  File "/mnt/data/home/stu/projects/external/pyglet/pyglet/window/__init__.py", line 726, in on_resize
    gl.glMatrixMode(gl.GL_PROJECTION)
```

see also   https://code.google.com/p/pyglet/issues/detail?id=532

This is probably a case where the binary nvidia driver is being a bit too forgiving.   It can be made to work by patching pyglets on_resize method - but it's not ideal.
